queens/iterators/control_variates_iterator.py

```diff
@@ -0,0 +1,246 @@
+                        * ``mean_cv`` (float): Mean of control variate.
+                        * ``std_cv`` (float): Standard deviation of control variate.
+                        * ``alpha`` (float): Method specific parameter that determines the
+                        influence of control variate.
+                        * ``sample_ratio`` (float): Ratio of number of samples on control variate to
+                                            number of samples on main model. Is only part of output
+                                            if use_optimal_num_samples==True.
+        num_samples_cv (int):    Number of samples to use for computing the expectation
+                                        value of control variate. If expectation of the control
+                                        variate is known, you can pass None as it's number of
+                                        samples.
+        samples (np.array): Samples for which all models are evaluated on.
+        use_optimal_num_samples (bool):    Determines if the iterator calculates and uses the
+                            optimal number of samples.
+        cost_model (float): Cost of evaluating the model.
+        cost_cv (float):    Cost of evaluating the control variate.
+        variance_cv_mean_estimator (float):   variance of the control variate mean estimator
+    """
+
+    @log_init_args
+    def __init__(
+        self,
+        model,
+        control_variate,
+        parameters,
+        global_settings,
+        seed,
+        num_samples,
+        expectation_cv=None,
+        num_samples_cv=None,
+        use_optimal_num_samples=False,
+        cost_model=None,
+        cost_cv=None,
+    ):
+        """Initialize the control variates iterator.
+
+        Args:
+            model (Model):  The main model. Uncertainties are quantified for this model.
+            control_variate (Model):    The control variate - a model that approximates the main
+            model and is cheaper to evaluate.
+            parameters (Parameters):    Parameters to use for evaluation.
+            global_settings (GlobalSettings):   Global settings to use for evaluation.
+            seed (int): Seed for random samples.
+            num_samples (int):  Number of samples to evaluate for estimation.
+            expectation_cv (float): Here you can pass expected values of the control variate.
+            num_samples_cv (int):   Number of MC samples to use for computing the expectation of
+                                    the control variate. Only used if the expectation of the control
+                                    variate is not passed.
+            cost_model (float): Cost of evaluating the model.
+            cost_cv (float):    Cost of evaluating the control variate.
+            use_optimal_num_samples (bool):  Determines if the iterator calculates and uses
+                                                the optimal number of samples.
```

Comment:
```suggestion
            use_optimal_num_samples (bool, opt): Determines whether the iterator calculates and uses the
                                                 optimal number of samples to estimate the control variate mean such that
                                                 the variance of the control variates estimator is minimized.
```
Also, please move this before `cost_model` to stick to the order of the arguments in the init function definition.